tests/integration/device_test.sh

```diff
@@ -4,87 +4,223 @@ include './tests/unit/utils.sh'
+  local output_cpu_max_freq
+  local output_cpu_min_freq
+
+  for distro in "${DISTROS[@]}"; do
+    container="kw-${distro}"
+
+    expected_cpu_model=$(container_exec "$container" "lscpu | grep 'Model name:' | cut --delimiter ':' --fields 2 | tr --squeeze-repeats ' ' | cut --delimiter ' ' --fields 2")
+    output_cpu_model=$(printf "%s" "${DEVICE_INFO_RESULTS[$distro]}" | grep --after-context 5 '^CPU:' | grep 'Model:' | cut --delimiter ':' --fields 2 | tr --delete ' ')
+    assert_equals_helper "'kw device' CPU model mismatch for ${distro}" "$LINENO" "$expected_cpu_model" "$output_cpu_model"
+
+    expected_cpu_max_freq=$(container_exec "$container" "lscpu | grep 'CPU max MHz' | cut --delimiter ':' --fields 2 | xargs")
+    output_cpu_max_freq=$(printf "%s" "${DEVICE_INFO_RESULTS[$distro]}" | grep 'Max frequency (MHz):' | cut --delimiter ':' --fields 2 | xargs)
+    assert_equals_helper "'kw device' CPU max frequency mismatch for ${distro}" "$LINENO" "$expected_cpu_max_freq" "$output_cpu_max_freq"
+
+    expected_cpu_min_freq=$(container_exec "$container" "lscpu | grep 'CPU min MHz' | cut --delimiter ':' --fields 2 | xargs")
+    output_cpu_min_freq=$(printf "%s" "${DEVICE_INFO_RESULTS[$distro]}" | grep 'Min frequency (MHz):' | cut --delimiter ':' --fields 2 | xargs)
+    assert_equals_helper "'kw device' CPU min frequency mismatch for ${distro}" "$LINENO" "$expected_cpu_min_freq" "$output_cpu_min_freq"
+  done
+}
+
+# Test the GPU information reported by 'kw device --local'
+function test_kw_device_local_gpu()
+{
+  local distro
+  local container
+  local pci_addresses
+  local gpu_info
+  local expected_gpu_model
+  local expected_gpu_provider
+  local output_gpu_model
+  local output_gpu_provider
+  declare -A expected_gpus
+
+  for distro in "${DISTROS[@]}"; do
+    container="kw-${distro}"
+
+    pci_addresses=$(container_exec "$container" "lspci | grep --regexp=VGA --regexp=Display --regexp=3D | cut --delimiter=' ' --fields 1")
+    for g in $pci_addresses; do
+      gpu_info=$(container_exec "$container" "lspci -v -s ${g}")
+      expected_gpu_model=$(printf '%s\n' "${gpu_info}" | sed --quiet --regexp-extended '/Subsystem/s/\s*.*:\s+(.*)/\1/p' | xargs)
+      expected_gpu_provider=$(printf '%s\n' "${gpu_info}" | sed --quiet --regexp-extended '/controller/s/.+controller: *([^\[\(]+).+/\1/p' | xargs)
+      expected_gpus["$g"]="${expected_gpu_model};${expected_gpu_provider}"
+    done
+
+    # Capture all lines starting from 'GPU:' until the next empty line
+    output_gpu=$(printf "%s" "${DEVICE_INFO_RESULTS[$distro]}" | sed -n '/^GPU:/,/^$/p')
+
+    for g in "${!expected_gpus[@]}"; do
+      expected_gpu_model="Model: $(printf '%s' "${expected_gpus[$g]}" | cut --delimiter ';' --fields 1)"
+      expected_gpu_provider="Provider: $(printf '%s' "${expected_gpus[$g]}" | cut --delimiter ';' --fields 2)"
+
+      # Check GPU model
+      printf "%s" "$output_gpu" | grep -q "$expected_gpu_model"
+      if [ $? -ne 0 ]; then
```

Comment:
Please don't use this style of `if`; this is not the default to the project. We prefer `[[ ]]`.